### PR TITLE
Add install and import guide for Python SDK

### PR DESCRIPTION
**Pull request by ojalatodd**
Added a short install and import tutorial for the
Python SDK. Each SDK should have instructions
on how to get the

### CODE REVIEW

**Comment by ojalatodd** [comment]
Note to reviewers: this PR contains commits from a non-merged PR, (#403), since it was branched off that PR. Not sure if there is a way to avoid having this extra commits in here.

File: docs/source/_templates/sdk_TP_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+Transaction Processor Tutorial  {{ language }}
+***********************************************
+
+Overview
+========
+
+This tutorial covers the creation of a new Sawtooth Lake transaction family in
+{{ language }}, based on the Sawtooth Lake SDK. We will construct a transaction
+handler which implements a distributed version of the multi-player game tic-
+tac-toe
```
Comment (aludvik) [inline docs/source/_templates/sdk_TP_tutorial.rst]: "."

File: docs/source/_templates/sdk_TP_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+.. note::
+
+    The SDK contains a fully-implemented version of tic-tac-toe. This tutorial
+    is meant to demonstrate the relevant concepts, rather than to create a
+    complete implementation. See the SDK_ for full implemenations in
+    multiple languages.
+
+.. _SDK: https://github.com/hyperledger/sawtooth-core/tree/master/sdk
```
Comment (aludvik) [inline docs/source/_templates/sdk_TP_tutorial.rst]: Maybe link to the examples folder?

File: docs/source/_templates/sdk_TP_tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+available in :doc:`/app_developers_guide/getting_started`.
+
+The Transaction Processor
+=========================
+
+There are two top-level components of a transaction processor: a processor
+class and a handler class. The SDK provides a general-purpose processor class.
+The handler class is application-dependent and contains the business logic for
+a particular family of transactions. Multiple handlers can be connected to a
+processor class.
```
Comment (aludvik) [inline docs/source/_templates/sdk_TP_tutorial.rst]: "an instance of the processor class."

File: docs/source/_templates/sdk_TP_tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+
+Handlers get called in two ways: an ``apply`` method and various metadata
```
Comment (aludvik) [inline docs/source/_templates/sdk_TP_tutorial.rst]: The Handler interface implements two types of methods:
1. An ``apply`` method
2. Various "metadata" methods

File: docs/source/_templates/sdk_TP_tutorial.rst [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+methods. The metadata is used to connect the handler to the processor, and
+we'll discuss it at the end of this tutorial. The bulk of the handler, however,
+is made up of ``apply`` and its helper functions, so that's where we'll start.
+
+The ``apply`` Method
+====================
+
+``apply`` gets called with two arguments, ``transaction`` and ``state_store``.
+``transaction`` holds the command that is to be executed (e.g. taking a space or
+creating a game), while ``state_store`` stores information about the current
```
Comment (aludvik) [inline docs/source/_templates/sdk_TP_tutorial.rst]: "while ``state_store`` encapsulates access to a snapshot of global validator state including information about...

(maybe link to http://intelledger.github.io/architecture/global_state.html?highlight=global?)

File: docs/source/_templates/sdk_TP_tutorial.rst [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+state of the game (e.g. the board layout and whose turn it is).
+
+Without yet getting into the details of how this information is encoded, we can
+start to think about what ``apply`` needs to do. ``apply`` needs to
```
Comment (aludvik) [inline docs/source/_templates/sdk_TP_tutorial.rst]: "start to think about what ``apply`` needs to do:"

File: docs/source/_templates/sdk_TP_tutorial.rst [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+
+1) unpack the command data from the transaction,
+2) retrieve the game data from the state store,
+3) play the game, and
+4) save the updated game data.
+
+Accordingly, a top-down approach to ``apply`` might look like this:
+
+{% if language == 'JavaScript' %}
```
Comment (aludvik) [inline docs/source/_templates/sdk_TP_tutorial.rst]: Do we have a story to fill this in? For now, maybe put something like "coming soon"

File: docs/source/_templates/sdk_TP_tutorial.rst (hunk 1 above)
Comment (ojalatodd) [inline docs/source/_templates/sdk_TP_tutorial.rst]: Fixed

File: docs/source/_templates/sdk_TP_tutorial.rst (hunk 2 above)
Comment (ojalatodd) [inline docs/source/_templates/sdk_TP_tutorial.rst]: Changed

File: docs/source/_templates/sdk_TP_tutorial.rst (hunk 3 above)
Comment (ojalatodd) [inline docs/source/_templates/sdk_TP_tutorial.rst]: Fixed

File: docs/source/_templates/sdk_TP_tutorial.rst (hunk 4 above)
Comment (ojalatodd) [inline docs/source/_templates/sdk_TP_tutorial.rst]: Changed to suggestion

Comment (ojalatodd) [inline docs/source/_templates/sdk_TP_tutorial.rst]: Changed to suggestion

File: docs/source/_templates/sdk_TP_tutorial.rst (hunk 7 above)
Comment (ojalatodd) [inline docs/source/_templates/sdk_TP_tutorial.rst]: It won't appear in the TOC right now anyway. And yes, there is a story!

Comment (ojalatodd) [inline docs/source/_templates/sdk_TP_tutorial.rst]: It won't appear in the TOC right now anyway. And yes, there is a story!

File: docs/source/_templates/sdk_TP_tutorial.rst [hunk 8]
```diff
@@ -0,0 +1,382 @@
...
+
+``/project/sawtooth-core/sdk/examples/xo_{{ lowercase_lang }}/``.
+
+Prerequisites
+=============
+
+This tutorial assumes that you have gone through :doc:`/app_developers_guide/getting_started` and are
+familiar with the concepts introduced there.
+
+Prior to going through this tutorial, you should have a working vagrant
```
Comment (rbuysse) [inline docs/source/_templates/sdk_TP_tutorial.rst]: Capitalize Vagrant

File: docs/source/_templates/sdk_TP_tutorial.rst (hunk 5 above)
Comment (aludvik) [inline docs/source/_templates/sdk_TP_tutorial.rst]: Did you address this?

File: docs/source/_templates/sdk_TP_tutorial.rst (hunk 6 above)
Comment (aludvik) [inline docs/source/_templates/sdk_TP_tutorial.rst]: Did you address this?

**Comment by ojalatodd** [comment]
@aludvik you are right about the complication -it's branched off another branch I have ready to merge with another PR.

**Comment by ojalatodd** [comment]
@aludvik By mistake, the most important file for this PR was not added last Friday: python_sdk_install.rst
I'll have to ask you to re-review.

File: docs/source/_templates/sdk_TP_tutorial.rst [hunk 9]
```diff
@@ -0,0 +1,382 @@
...
+
+There are two top-level components of a transaction processor: a processor
+class and a handler class. The SDK provides a general-purpose processor class.
+The handler class is application-dependent and contains the business logic for
+a particular family of transactions. Multiple handlers can be connected to an
+instance of the processor class.
+
+Handlers get called in two ways:
+
+# An ``apply`` method
```
Comment (rbuysse) [inline docs/source/_templates/sdk_TP_tutorial.rst]: This list isn't rendering correctly.

Comment (rbuysse) [inline docs/source/_templates/sdk_TP_tutorial.rst]: https://build.sawtooth.me/job/Sawtooth-Hyperledger/job/sawtooth-core/job/PR-422/6/artifact/docs/build/html/_autogen/sdk_TP_tutorial_python.html#the-transaction-processor

**Comment by ojalatodd** [comment]
This PR is being replaced by PR #431

**Comment by ojalatodd** [comment]
Closing in favor of #431

**Pull request closed without merging**